### PR TITLE
Guild mod hooks

### PR DESCRIPTION
**Pull request by ajrb**
1. Fixes to guild foundations.
2. Guild mod hooks:
2.1 Building replacement functionality in WorldDataReplacement reads building json from loose files or mods (also caches it when not running in editor in a non-sparse cache so abort calls early since block data is loaded a LOT - please check the code here @Interkarma & @TheLacus )
2.2 Register new custom factions, guilds, and services from mods
3. Fix NPC nameseeds
4. Three dump data commands

To test this you will need to do the following (after checking it hasn't broken anything first) to add my custom guild mod:

1. Download the mod file from my mod repo (https://github.com/ajrb/dfunity-mods/raw/master/build/StandaloneWindows/archaeologists.dfmod)
2. Place in StreamingAssets/Mods and ensure is activated
3. Load game and travel to Oxbeth
4. Look at map and find the large irregularly shaped building (actually is 2 separate buildings) get close to one
5. Use info mode and click on building - should say Archaeologists guild
6. Enter both buildings and interact with NPCs
7 Grin! :-D

Let me know if you find any issues. Hope you have some time to look at it this weekend.

### CODE REVIEW

**Comment by Interkarma** [comment]
Very nice! 👍 

Will merge this to its own branch to start with, as that helps me test larger PRs in a sandbox. Will merge back to master as soon as possible though!

**Comment by ajrb** [comment]
Yes that's what I want you to do - put in a branch for testing and checking period. This is a big bunch o changes! While I am very proud of what I've pulled together, I would be surprised if I've got it all right.

**Comment by Interkarma** [comment]
No probs. :)

**Comment by TheLacus** [comment]
Hey this is great! I'll test it as soon as i can.

File: Assets/Scripts/Utility/AssetInjection/WorldDataReplacement.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                {
+                    buildingData = Buildings[blockRecordId];
+                    return (buildingData.BuildingType != noReplacementBT);
+                }
+                else
+                {
+                    string fileName = GetBuildingReplacementFilename(blockName, blockIndex, recordIndex);
+
+                    // Seek from loose files
+                    if (File.Exists(Path.Combine(worldDataPath, GetBuildingReplacementFilename(blockName, blockIndex, recordIndex))))
```
Comment (TheLacus) [inline Assets/Scripts/Utility/AssetInjection/WorldDataReplacement.cs]: You got the filename just above :)

Comment (ajrb) [reply Assets/Scripts/Utility/AssetInjection/WorldDataReplacement.cs]: Whoops, good catch.

**Comment by ajrb** [comment]
Glad you like it, look forward to any feedback.

**Pull request merged**